### PR TITLE
obfs4proxy: update to version 0.0.9

### PR DESCRIPTION
**Pull request by ja-pa**
Maintainer: @jefferyto
Compile tested: Turris Omnia (TOS4), OpenWrt 18.06.1
Run tested: Turris Omnia (TOS4), OpenWrt 18.06.1

Description:

Changes:
-remove old patch(part of 0.0.9 release)
-change the canonical upstream repo location to gitlab
-change source url to github.com

Part of the update is also the change of upstream URL as described here https://gitweb.torproject.org/pluggable-transports/obfs4.git/commit/?id=70d0e90c861be34ce3c5425ef1366a0b2ceb3026

Signed-off-by: Jan Pavlinec <jan.pavlinec@nic.cz>

### CODE REVIEW

File: net/obfs4proxy/Makefile [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
-PKG_VERSION:=0.0.7
+PKG_VERSION:=0.0.8
 PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_URL:=https://git.torproject.org/pluggable-transports/obfs4.git
-PKG_SOURCE_VERSION:=8256fac93c2cf79742725e3aaced5bbe3380fd32
-PKG_SOURCE_DATE:=20161115
-PKG_MIRROR_HASH:=9c504337f7a2b46310083224cf0998fc7645f8af5d50657cf74358d39916eddc
+PKG_SOURCE_URL:=https://gitlab.com/yawning/obfs4.git
```
Comment (diizzyy) [inline net/obfs4proxy/Makefile]: https://github.com/Yawning/obfs4
Use codeload

Comment (ja-pa) [reply net/obfs4proxy/Makefile]: ok, no problem. Is there any particular reason why is GitHub preferred over Gitlab ?

Comment (diizzyy) [reply net/obfs4proxy/Makefile]: Tarballs are preferred over git/svn etc

Comment (ja-pa) [reply net/obfs4proxy/Makefile]: changed to codeload

**Comment by ja-pa** [comment]
@diizzyy I changed source URL to codeload and updated version to latest release. I'll appreciate other suggestions. Thx

**Comment by diizzyy** [comment]
LGTM

**Comment by ja-pa** [comment]
@jefferyto What do you think?

**Comment by jefferyto** [comment]
@ja-pa Sorry, been busy with other things. I was holding out on reviewing this until after I had a chance to test compiling 0.0.8 (now 0.0.9) for myself.

I have a few changes I'd like to see:

* Change the GitHub codeload link to https://gitlab.com/yawning/obfs4/-/archive/obfs4proxy-0.0.9/obfs4-obfs4proxy-0.0.9.tar.bz2 .

  Aside from my concerns about codeload itself, I see two reasons to not use it in this case:
  1. The GitHub mirror is only a "courtesy" mirror. Yawning Angel (the software author) has [stated](https://gitlab.com/yawning/obfs4/commit/70d0e90c861be34ce3c5425ef1366a0b2ceb3026) that he reserves the right to stop pushing to GitHub at any time.
  2. GitLab has download links for [tagged commits](https://gitlab.com/yawning/obfs4/tags). While these may have the same problems as GitHub codeload, at least it is hosted at the official repo.

* Rename the source package from `golang-torproject-pluggable-transports-obfs4-dev` to `gitlab-yawning-obfs4-dev`.

  There are multiple references to this name in the Makefile. Ideally we would also add a [compatibility symlink](https://go-team.pages.debian.net/packaging.html#_upstream_package_moves) but I doubt anyone has ever actually installed or used the source package, so I think we can skip it.

* Remove the dependencies in the source package.

  (This is the change I wanted to test for myself.) Now that obfs4proxy has been [ported to a Go module](https://gitlab.com/yawning/obfs4/commit/08f4d470188e9dad8658f2d2107880d8fdf326be), it manages (and downloads) its own dependencies.

Thanks for your contribution!

**Comment by jefferyto** [comment]
@diizzyy Please consult the maintainer before telling contributors to use codeload.

**Comment by diizzyy** [comment]
@jefferyto 
The common goal is to lower traffic usage and have consistent tarballs. git/svn/hg is not the first option having that in mind.

**Comment by ja-pa** [comment]
@jefferyto I changed PR   according to your suggestion. Will be glad for other comments. Thanks

**Comment by jefferyto** [comment]
@ja-pa Thanks! Can you remove these lines:

```
  DEPENDS+= \
    +golang-github-agl-ed25519-dev \
    +golang-github-dchest-siphash-dev \
    +golang-golang-x-crypto-dev \
    +golang-golang-x-net-dev \
    +golang-torproject-pluggable-transports-goptlib-dev
```

from `define Package/golang-gitlab-yawning-obfs4-dev`? It should still compile correctly. (If not, please make sure you are using a recent snapshot sdk, and if it still doesn't work, maybe wait until #8181 is merged.)

**Comment by hnyman** [comment]
@jefferyto @ja-pa 
Are modifications still needed here?

**Comment by jefferyto** [comment]
@hnyman this can be merged - I can remove the unnecessary `DEPENDS` later.

**Pull request merged**